patches/minecraft/net/minecraft/tileentity/MobSpawnerBaseLogic.java.patch

```diff
@@ -5,11 +5,20 @@
                      entity.func_70012_b(entity.field_70165_t, entity.field_70163_u, entity.field_70161_v, world.field_73012_v.nextFloat() * 360.0F, 0.0F);
  
 -                    if (entityliving == null || entityliving.func_70601_bi() && entityliving.func_70058_J())
-+                    if (entityliving == null || net.minecraftforge.event.ForgeEventFactory.canEntitySpawnSpawner(entityliving, func_98271_a(), (float)entity.field_70165_t, (float)entity.field_70163_u, (float)entity.field_70161_v))
++                    if (entityliving == null || net.minecraftforge.event.ForgeEventFactory.canEntitySpawnSpawner(entityliving, func_98271_a(), (float)entity.field_70165_t, (float)entity.field_70163_u, (float)entity.field_70161_v, this))
                      {
                          if (this.field_98282_f.func_185277_b().func_186856_d() == 1 && this.field_98282_f.func_185277_b().func_150297_b("id", 8) && entity instanceof EntityLiving)
                          {
-+                            if (!net.minecraftforge.event.ForgeEventFactory.doSpecialSpawn(entityliving, this.func_98271_a(), (float)entity.field_70165_t, (float)entity.field_70163_u, (float)entity.field_70161_v))
++                            if (!net.minecraftforge.event.ForgeEventFactory.doSpecialSpawn(entityliving, this.func_98271_a(), (float)entity.field_70165_t, (float)entity.field_70163_u, (float)entity.field_70161_v, this))
                              ((EntityLiving)entity).func_180482_a(world.func_175649_E(new BlockPos(entity)), (IEntityLivingData)null);
                          }
  
+@@ -296,6 +297,8 @@
+ 
+     public abstract BlockPos func_177221_b();
+ 
++    @Nullable public Entity getSpawnerEntity() { return null; }
```

Comment:
This method should go at the end of the file, after a comment marking the forge section:
`/* ======================================== FORGE START =====================================*/`